gemma-web/src/main/java/ubic/gemma/web/controller/OntologyController.java

```diff
@@ -67,14 +75,36 @@ public RedirectView getTerm( @PathVariable("termId") String termId ) {
         if ( term == null ) {
             throw new EntityNotFoundException( String.format( "No term with IRI %s in TGEMO.", iri ) );
         }
-        String gemmaOntologyUrl = gemmaOntologyService.getOntologyUrl();
-        String urlWithFragment = UriComponentsBuilder.fromHttpUrl( gemmaOntologyUrl )
-                .fragment( iri )
-                .build()
-                .toUriString();
-        RedirectView redirectView = new RedirectView( urlWithFragment );
-        redirectView.setStatusCode( HttpStatus.FOUND );
-        return redirectView;
+
+        StringBuilder s = new StringBuilder( String.format( "<title>%s</title>", escapeHtml4( term.getLabel() ) ) +
+                "<div class=\"padded\"><ul style=\"padding-bottom:0.5em\">" +
+                String.format( "<h2 style=\"padding-bottom:0.5em\">%s: %s</h2>",
+                        escapeHtml4( term.getLocalName() ),
+                        escapeHtml4( term.getLabel() ) ));
+                if ( term.getUri() != null) {
+                    s.append(String.format( "<li><a href=%s>%s</a></li>",
+                            escapeHtml4(term.getUri()
+                                    .replaceFirst( "^" + Pattern.quote( TGEMO_URI_PREFIX ), servletContext.getContextPath() + "/ont/" ) ) ,
+                            escapeHtml4( TGEMO_URI_PREFIX + term.getLocalName() ) ) );
+                }
+
+
+        for (AnnotationProperty annot: term.getAnnotations()) {
+            if (Arrays.asList("hasDefinition","definition").contains(annot.getProperty()) ){
+                s.append( "<li>" )
+                        .append( annot.getContents() )
+                        .append( "</li>" );
+            }
+        }
+
+
+        s.append( "</ul>" );
+        s.append( String.format("<p>Ontology IRI: <a href=%s>%s</a></p>",
+                servletContext.getContextPath() + "/ont/TGEMO",
+                TGEMO_URI_PREFIX + "TGEMO" ) );
+        s.append( "</div>" );
+        return s.toString();
+
```

Comment:
Also, as an alternative, you can also create a JSP page and pass the variables you need over there. This can wait for another iteration though.